specutils/analysis/width.py

```diff
@@ -0,0 +1,212 @@
+import numpy as np
+from astropy.stats.funcs import gaussian_fwhm_to_sigma
+from ..spectra import SpectralRegion
+
+
+__all__ = ['gaussian_sigma_width', 'gaussian_fwhm', 'fwhm']
+
+
+def _computation_wrapper(func, spectrum, region):
+
+    # No region, therefore whole spectrum.
+    if region is None:
+        return func(spectrum)
+
+    # Single region
+    elif isinstance(region, SpectralRegion):
+        return func(spectrum, region=region)
+
+    # List of regions
+    elif isinstance(region, list):
+        return [func(spectrum, region=reg) for reg in region]
+
+
+def gaussian_sigma_width(spectrum, region=None):
+    """
+    Estimate the full width of the spectrum based on an approximate value of
+    sigma.  This will be calculated over the regions, if they are specified.
```

Comment:
Might reword this a bit to something like "Estimate the width of the spectrum using a second-moment analysis, scaled to match the sigma/standard deviation parameter of a standard Gaussian profile."? I'm not sure the grammar is sound in what I wrote there, but I think it at least hits on all the key mathematical points... 😉